### PR TITLE
worked on pet sitter schema with proposed discriminator

### PR DESCRIPTION
**Pull request by MUGABA**
### What this PR does (required):
- This Pr will create a pet sitter schema

### CODE REVIEW

**Comment by ezeilo-su** [comment]
Hey @MUGABA, please link the ticket for this PR before we can proceed with review

File: server/models/PetSitter.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+  },
+  activatedAvailabilitySchedule: {
+    type: mongoose.Schema.Types.ObjectId,
+    required: true,
+    ref: "Schedule",
+  },
+  rate: {
+    type: Number,
+    required: true,
+  },
```
Comment (ugur-akin) [inline server/models/PetSitter.js:22]: It would be cool to validate that this is a "reasonable" number! ;)

File: server/models/Profile.js [hunk 2]
```diff
@@ -28,12 +28,13 @@ const profileSchema = new mongoose.Schema({
...
   birthday: {
     type: Date,
-    default: null
+    default: null,
   },
   photo: {
     type: String,
     default: "",
   },
+  discriminatorKey: "userType",
```
Comment (ugur-akin) [inline server/models/Profile.js:37]: I believe this should be passed as an [option](https://mongoosejs.com/docs/guide.html#options) (here it's just another field in the model).

**Review by ugur-akin — CHANGES_REQUESTED** [review]
Good work, almost there 👍🏼

File: server/models/PetSitter.js (hunk 1 above)
Comment (MUGABA) [reply server/models/PetSitter.js:22]: I have changed something. Please let me know if that is what we need!

File: server/models/Profile.js [hunk 3]
```diff
@@ -1,39 +1,46 @@
...
-    type: Date,
-    default: null
-  },
-  photo: {
-    type: String,
-    default: "",
-  },
-});
+const options = {
+  discriminatorKey: "userType",
```
Comment (moffatethan) [inline server/models/Profile.js:4]: Let's just call this `type`

**Review by moffatethan — CHANGES_REQUESTED** [review]

**Review by moffatethan — APPROVED** [review]

**Pull request merged**